src/cuda/utils.rs

```diff
@@ -11,6 +11,7 @@ use crate::error::{GPUError, GPUResult};
 // are never used directly, they are only accessed through [`cuda::Device`] which contains an
 // `UnownedContext`. A device cannot have an own context itself, as then it couldn't be cloned,
 // but that is needed for creating the kernels.
+#[allow(dead_code)]
 pub(crate) struct CudaContexts(Vec<rustacuda::context::Context>);
```

Comment:
This is really an annoying one, though I'd prefer:

```rust
pub(crate) struct CudaContexts(#[allow(unused)] Vec<rustacuda::context::Context>);
```

As it isn't really dead code.